### PR TITLE
Technical bootstrap

### LINKED ISSUE #2: Technical bootstrap

### PR DESCRIPTION
**Pull request by AbdelrhmanHamouda**
close #2

Implement:
- Dockerization 
- push to dockerHub on tag (i.e. release)
- build & test docs as part of CI flow 
- Publish docs on tag (i.e. release)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,8 +43,28 @@ jobs:
       - name: Test package
         run: docker run --rm ${{ env.TEST_IMAGE }} "poe test"
 
-      # TODO setup coverage in #8
-#      - name: Run codacy coverage reporter
-#        uses: codacy/codacy-coverage-reporter-action@v1
-#        with:
-#          project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
+  docs-test:
+    name: Test documentation
```
Comment (nmercier79) [inline .github/workflows/ci.yml:47]: is it the documention of Tests or the test of documentation ? in the steps I see mkdocs build "only" does it include running tests ?

File: poetry.lock [hunk 2]
```diff
@@ -127,7 +127,7 @@ optional = false
 python-versions = ">=3.6.0"
 
 [package.extras]
-unicode-backport = ["unicodedata2"]
+unicode_backport = ["unicodedata2"]
```
Comment (nmercier79) [inline poetry.lock:130]: for my knowledge I see you replaced multiple package naming from - to _ is it a new issue, a coding convention or something ?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -43,8 +43,28 @@ jobs:
...
+    runs-on: ubuntu-latest
```
Comment (bearfoot) [inline .github/workflows/ci.yml:48]: So steps are github actions, right?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (AbdelrhmanHamouda) [reply .github/workflows/ci.yml:47]: it is a step that checks "tests" that documentation site will build without an issue.  Thus, the `build` command if it runs without errors, it means that the docs will be able to be published for the gh-pages branch

File: poetry.lock (hunk 2 above)
Comment (AbdelrhmanHamouda) [reply poetry.lock:130]: anything in the .lock file is completely automated and managed by `poetry`. So my guess is that they are doing something with the naming.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (AbdelrhmanHamouda) [reply .github/workflows/ci.yml:48]: Steps are like jenkins stages and jobs are like jenkins pipelines and workflow is a collection of jobs based on a specific trigger.  

One can run a Github action in a step but i also think that they call the whole CI system "GitHub Actions"

**Review by nmercier79 — APPROVED** [review]

**Pull request merged**